### DISCUSSION TITLE
Can I add zeros after last frame?

### DISCUSSION THREAD

**Discussion by NextBG** [sist-cs120/project-wiki#19] [Project 1 Q&A] [answered]
If the length of the input is not divisible by my frame length, can I add zeros after the last frame to make its length the same as my frame length?

For example, my frame length is 8 bits
INPUT.txt is: 11111111 1111
OUTPUT.txt is: 11111111 11110000

**Comment by yhyan96** [comment, marked as answer]
You can add zeros in the frame.

But you must remove them when you decode the frame.

It is recommended to add some bits to present the  length of real data, between the header and the data.

For example, the frame length is 3 bits (data length) + 8 bits (data)
INPUT.txt is: 11111111 1111
Frame 1 is: Header + 111 + 11111111
Frame 2 is: Header + 011 + 11110000
Now you can know that Frame 2 contains 4 real bits and the others are just padding.
OUTPUT.txt is: 11111111 1111
